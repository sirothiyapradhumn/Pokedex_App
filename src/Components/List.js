import React, { useEffect, useState } from 'react'
import PokeCard from './PokeCard';

function List({currText, pokeType, pokeWeakness, data}) {
    const [pokeData, setPokedata] = useState([]);
    // console.log(currText);
    //console.log(pokeType);
    //console.log(pokeWeakness);

    const fetchData = async () => {
      try {
        const response = await fetch(`https://raw.githubusercontent.com/Biuni/PokemonGO-Pokedex/master/pokedex.json`);
        const json = await response.json();
        //console.log(json.pokemon);
        setPokedata(json.pokemon);
      } catch (error) {
        console.log("error", error);
      }
    };

    useEffect(()=>{
        // console.log("component did mount");
        // (async () => {
        //     let ans = await axios.get(`https://raw.githubusercontent.com/Biuni/PokemonGO-Pokedex/master/pokedex.json`);
        //     console.log(ans.data.pokemon);
        //     setPokedata(ans.data.pokemon);
        //   })();
        
        fetchData();

        return () => {
        // this now gets called when the component unmounts
        };
      },[])

      
			
      function arraysEqual(a, b) {
        if (a === b) return true;
        if (a == null || b == null) return false;
        if (a.length < b.length) return false;
       
        for (var i = 0; i < b.length; ++i) {
          if(a.includes(b[i]) == false){
            return false;
          }
        }
        return true;
      }
      
      var filteredPokeByTypes = pokeData.filter( (obj)=>{
        return (arraysEqual(obj.type.sort(), pokeType.sort()));
      });

      var filteredPokeByWeakness = filteredPokeByTypes.filter((obj)=>{
        return (arraysEqual(obj.weaknesses.sort(), pokeWeakness.sort()));
      });

      //console.log(filteredPokeByTypes);
      //console.log(filteredPokeByWeakness);

    let filteredPoke =  filteredPokeByWeakness;
    //console.log(filteredPoke);

    if(currText === ""){
      filteredPoke = filteredPokeByWeakness;
    }
    else{
      filteredPoke = filteredPoke.filter((poke) =>{
            let pokeName = poke.name.toLowerCase();
            return pokeName.includes(currText); 
        });
    }  

  return (
    <div className='all-cards'>
      {filteredPoke.map((row) => (<PokeCard row={row} data={data}/>))}
    </div>
  )
}

export default List