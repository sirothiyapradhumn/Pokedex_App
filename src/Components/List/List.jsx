/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react'
import PokeCard from '../PokeCard/PokeCard';
import style from './list.module.scss'

function List({currText, pokeType, pokeWeakness, data, apiCall}) {
    const [pokeData, setPokedata] = useState([]);

    const fetchData = async () => {
      try {
        const response = await fetch(`https://raw.githubusercontent.com/Biuni/PokemonGO-Pokedex/master/pokedex.json`);
        const json = await response.json();
        apiCall(json.pokemon);
        setPokedata(json.pokemon);
      } catch (error) {
        console.log("error", error);
      }
    };

    useEffect(()=>{
        fetchData();
        return () => {
        // this now gets called when the component unmounts
        };
      },[])

      
			
      function arraysEqual(a, b) {
        if (a === b) return true;
        if (a == null || b == null) return false;
        if (a.length < b.length) return false;
        for (var i = 0; i < b.length; ++i) {
          if(a.includes(b[i]) === false){
            return false;
          }
        }
        return true;
      }
      
      var filteredPokeByTypes = pokeData.filter( (obj)=>{
        return (arraysEqual(obj.type.sort(), pokeType.sort()));
      });

      var filteredPokeByWeakness = filteredPokeByTypes.filter((obj)=>{
        return (arraysEqual(obj.weaknesses.sort(), pokeWeakness.sort()));
      });


    let filteredPoke =  filteredPokeByWeakness;

    if(currText === ""){
      filteredPoke = filteredPokeByWeakness;
    }
    else{
      filteredPoke = filteredPoke.filter((poke) =>{
            let pokeName = poke.name.toLowerCase();
            return pokeName.includes(currText); 
        });
    }  

  return (
    <div className={style.allCards}>
      {filteredPoke.map((row) => (<PokeCard row={row} data={data}/>))}
    </div>
  )
}

export default List