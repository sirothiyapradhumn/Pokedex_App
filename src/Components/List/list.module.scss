.allCards {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    align-items: center;
    background-color: #1f2833;
  }