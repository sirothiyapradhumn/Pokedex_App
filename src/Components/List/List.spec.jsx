import React from 'react'
import { shallow, mount } from "enzyme";
import List from './List'

const unmockedFetch = global.fetch



describe('List component', ()=>{
    let mockPt = ['Grass', 'Poison'];
    let mockPw = ['Fire'];
    let mockCt = 'b';

    beforeAll(() => {
        global.fetch = () =>
          Promise.resolve({
            json: () => Promise.resolve([]),
          })
      })

    it('testing List coverage', async()=>{
        let wrapper = mount(<List currText={mockCt} pokeType={mockPt} pokeWeakness={mockPw} />);
        expect(wrapper).toMatchSnapshot();
    })
    

    afterAll(() => {
        global.fetch = unmockedFetch
      })

})