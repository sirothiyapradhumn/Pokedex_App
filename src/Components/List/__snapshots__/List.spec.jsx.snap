// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component testing List coverage 1`] = `
<List
  currText="b"
  pokeType={
    Array [
      "Grass",
      "Poison",
    ]
  }
  pokeWeakness={
    Array [
      "Fire",
    ]
  }
>
  <div
    className="allCards"
  />
</List>
`;
